### PR TITLE
doc: Update bips.md for 23.x

### PR DESCRIPTION
**Pull request by laanwj**
As far as I know, there have been no new bips implemented in this major release. Update `bips.md` accordingly.
(if there are, please post below)

### CODE REVIEW

**Comment by MarcoFalke** [comment]
cr ACK e97e3ded69ba1341154bbbea0c75bfe6c09c02e0

**Review by jonatack — COMMENTED** [review]
ACK e97e3ded69ba1341154bbbea0c75bfe6c09c02e0 if no BIPs were newly implemented in v23

**Comment by fanquake** [comment]
sipa on IRC:
> Heh, bips.md does not list BIP157 which we support since 0.21.0.

File: doc/bips.md [hunk 1]
```diff
@@ -42,7 +42,8 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v22.0**):
...
 * [`BIP 152`](https://github.com/bitcoin/bips/blob/master/bip-0152.mediawiki): Compact block transfer and related optimizations are used as of **v0.13.0** ([PR 8068](https://github.com/bitcoin/bitcoin/pull/8068)).
 * [`BIP 155`](https://github.com/bitcoin/bips/blob/master/bip-0155.mediawiki): The 'addrv2' and 'sendaddrv2' messages which enable relay of Tor V3 addresses (and other networks) are supported as of **v0.21.0** ([PR 19954](https://github.com/bitcoin/bitcoin/pull/19954)).
-* [`BIP 158`](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki): Compact Block Filters for Light Clients can be indexed as of **v0.19.0** ([PR #14121](https://github.com/bitcoin/bitcoin/pull/14121)).
+* [`BIP 157`](https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki)
+  [`158`](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki): Compact Block Filters for Light Clients can be indexed as of **v0.19.0** ([PR #14121](https://github.com/bitcoin/bitcoin/pull/14121)), and served to peers on the P2P network as of **v0.21.0** ([PR #16442](https://github.com/bitcoin/bitcoin/pull/16442)).
```
Comment (jonatack) [inline doc/bips.md:46]: I wondered if #16442 should be #18876 but this seems right. The PR chain is somewhat circuitous.

**Review by jonatack — COMMENTED** [review]
ACK 8472b77e6b7bf822b4baf7a19adf95203d178231

File: doc/bips.md (hunk 1 above)
Comment (laanwj) [reply doc/bips.md]: Updated

**Comment by jonatack** [comment]
ACK 03bc08e16325f43905f6e6f8d5e0ce69aa8a30a4

**Review by ghost — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/24420/commits/03bc08e16325f43905f6e6f8d5e0ce69aa8a30a4

File: doc/bips.md [hunk 2]
```diff
@@ -42,7 +42,8 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v22.0**):
...
 * [`BIP 152`](https://github.com/bitcoin/bips/blob/master/bip-0152.mediawiki): Compact block transfer and related optimizations are used as of **v0.13.0** ([PR 8068](https://github.com/bitcoin/bitcoin/pull/8068)).
 * [`BIP 155`](https://github.com/bitcoin/bips/blob/master/bip-0155.mediawiki): The 'addrv2' and 'sendaddrv2' messages which enable relay of Tor V3 addresses (and other networks) are supported as of **v0.21.0** ([PR 19954](https://github.com/bitcoin/bitcoin/pull/19954)).
-* [`BIP 158`](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki): Compact Block Filters for Light Clients can be indexed as of **v0.19.0** ([PR #14121](https://github.com/bitcoin/bitcoin/pull/14121)).
+* [`BIP 157`](https://github.com/bitcoin/bips/blob/master/bip-0157.mediawiki)
+  [`158`](https://github.com/bitcoin/bips/blob/master/bip-0158.mediawiki): Compact Block Filters for Light Clients can be indexed as of **v0.19.0** ([PR #14121](https://github.com/bitcoin/bitcoin/pull/14121)) and served to peers on the P2P network as of **v0.21.0** ([PR #16442](https://github.com/bitcoin/bitcoin/pull/16442)).
```
Comment (gruve-p) [inline doc/bips.md:46]: `158` > ` BIP158`

Comment (prayank23) [reply doc/bips.md:46]: `BIP 157` `158`

It's same as `BIP 340` `341 ` `342`

Comment (michaelfolkson) [reply doc/bips.md:46]: Right, the doc doesn't repeat "BIP" when there are multiple BIP numbers on the same line.

**Pull request merged**